velox/exec/RowNumber.cpp

```diff
@@ -444,6 +419,23 @@ void RowNumber::spill() {
     spillInput(input_, memory::spillMemoryPool());
     input_ = nullptr;
   }
+
+  // Recursive spill is triggered.
+  if (spillInputReader_ != nullptr) {
```

Comment:
This will take very long and we probably can't run this step during spill which run under memory arbitration.